File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    answer = ''
    memo = {}
    for p in participant:
        if p in memo:
            memo[p] += 1
        else:
            memo[p] = 1
            
    for c in completion:
        memo[c] -= 1
        
    for key, value in memo.items():
        if value == 1:
            answer = key
    return answer

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        Map<String, Integer> memo = new HashMap<>();
        
        for (String p : participant) {
            if (memo.containsKey(p)) {
                memo.put(p, memo.get(p) + 1);
            } else {
                memo.put(p, 1);
            }
        }
        
        for (String c : completion) {
            memo.put(c, memo.get(c) - 1);
        }
        
        for (String p : participant) {
            if (memo.get(p) == 1) {
                answer = p;
            }
        }
        
        return answer;
    }
}